### ISSUE TITLE
vcl.inline always shows the same error (VCL compilation failed)

### ISSUE THREAD

**Issue by coudenysj** [varnishcache/varnish-cache#2461]
When I use vcl.inline to test/load a VCL I got the error "VCL compilation failed", but I did not receive where the error in the VCL was.

After some investigation (and help on IRC) I found out Varnish was not able to store the tmp VCL file in its current working directory (because it did not exist anymore).

## Expected Behavior

Show a more descriptive error message.

## Current Behavior

The same "VCL compilation failed" message as when the VCL contains errors.

Strace output:
```
mkdir("vcl_jachimtest.1507897645.070521355", 0755) = -1 ENOENT (No such file or directory)
write(2, "Error: Cannot create VCL directory 'vcl_jachimtest.1507897645.070521355': No such file or directory\n", 100) = 100
sendto(5, "<131>Oct 13 14:27:25 varnishd/testvar[3837]: Cannot create VCL directory 'vcl_jachimtest.1507897645.070521355': No such file or directory", 137, MSG_NOSIGNAL, NULL, 0) = 137
sendto(5, "<134>Oct 13 14:27:25 varnishd/testvar[3837]: CLI telnet 178.208.60.54 42846 178.208.60.61 10003 Wr 106 VCL compilation failed", 125, MSG_NOSIGNAL, NULL, 0) = 125
writev(7, [{"106 22      \n", 13}, {"VCL compilation failed", 22}, {"\n", 1}], 3) = 36
```

## Possible Solution

Test if the CWD is writeable before using it (and printing that message as a reponse).

## Steps to Reproduce (for bugs)
1. Start vanishd instance with /tmp/test as CWD (-n)
2. Delete the /tmp/test directory
3. Request vcl.inline via telnet or varnishadm

## Your Environment
* Version used: varnish-4.1.8 revision d266ac5c6
* Operating System and version: Linux,3.14.51-20150818-grsec,x86_64,-junix,-smalloc,-smalloc,-hcritbit

**Comment by Dridi** [comment]
Hello, thanks for reporting this. To improve the error message, we should probably add `errno`/`strerror` to the CLI response when we don't have an error from the compiler. If we add a check, there's no guarantee that it will hold true afterwards, so we rather want to report errors as they happen.

**Comment by bsdphk** [comment]
Fixed in f16eed37949ab298abb83bc59ea59b444fe4e07f

**Comment by coudenysj** [comment]
![giphy 6](https://user-images.githubusercontent.com/96260/36729203-b09dd1bc-1bc3-11e8-9923-d96ef9d50cf5.gif)
